### ISSUE TITLE
KeyError: 'attributes' on ldap_sync_users

### ISSUE THREAD

**Issue by codeshard** [etianen/django-python3-ldap#31]
I'm trying to sync all my active directory users to my local django users, all attempts has been unsuccessful, the command gives me a *KeyError: 'attributes' on attributes = user_data["attributes"]*. Also the login doesn't work.
I'm using Django 1.8.6 with Python 3.4, here's my ldap settings:

    LDAP_AUTH_URL = "ldaps://10.0.0.1:636"
    LDAP_AUTH_USE_TLS = None
    LDAP_AUTH_SEARCH_BASE = "DC=mydomain,DC=example,DC=com"
    LDAP_AUTH_OBJECT_CLASS = "organizationalPerson"
    LDAP_AUTH_USER_FIELDS = {
        "username": "sAMAccountName",
        "first_name": "givenName",
        "last_name": "sn",
        "email": "mail",
    }

    LDAP_AUTH_USER_LOOKUP_FIELDS = ("username",)

    LDAP_AUTH_CLEAN_USER_DATA = "django_python3_ldap.utils.clean_user_data"

    LDAP_AUTH_SYNC_USER_RELATIONS = "django_python3_ldap.utils.sync_user_relations"

    LDAP_AUTH_FORMAT_SEARCH_FILTERS = "django_python3_ldap.utils.format_search_filters"

    LDAP_AUTH_FORMAT_USERNAME =     "django_python3_ldap.utils.format_username_active_directory"

    LDAP_AUTH_CONNECTION_USERNAME =   'CN=admin,OU=Users,OU=MyGroup,OU=MyDomain,DC=mydomain,DC=example,DC=com'

    LDAP_AUTH_CONNECTION_PASSWORD = 'f4ncyp4ssw0rd'

    AUTHENTICATION_BACKENDS = (
        'django_python3_ldap.auth.LDAPBackend',
        'django.contrib.auth.backends.ModelBackend',
    )

I modified the ldap.py module printint the value of user_data and gives me this output:

    {'uri': ["b'ldaps://mydomain.example.com/CN=Configuration,DC=mydomain,DC=example,DC=com'"], 'type': 'searchResRef'}

So, the dictionary cames without the attributes value, what I am missing? Configuration error?

PS: Sorry by me terrible english ;)

**Comment by etianen** [comment]
Try using the master branch. It's got a small change that might help debug
this.

**Comment by codeshard** [comment]
Thanks for your answer! I'm using the master branch right now, I've tested the pip version as well. The error I got above was my own fault, 'cause I'm using a MS AD and I must specify a OU in the LDAP_AUTH_SEARCH_BASE setting, like this:
    LDAP_AUTH_SEARCH_BASE = "OU=People,DC=mydomain,DC=example,DC=com"
Now django-python3-ldap works like a charm! 
Closing the issue 'cause there's not issue at all.
Thanks once again!
